### PR TITLE
[AArch64][GlobalISel] Legalize G_ABS for Larger/Smaller Vectors

### PR DESCRIPTION
**Pull request by chuongg3**
Legalize G_ABS for larger/smaller width vectors with legal element sizes

Fallsback for the smaller width vector tests because it is unable to legalize for G_ANYEXT smaller width vectors

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/abs.ll [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    ret <8 x i8> %3
+}
+declare <8 x i8> @llvm.abs.v8i8(<8 x i8>, i1)
+
+define <16 x i8> @abs_v16i8(<16 x i8> %0){
+; CHECK-LABEL: abs_v16i8:
+; CHECK:       // %bb.0:
+; CHECK-NEXT:    abs v0.16b, v0.16b
+; CHECK-NEXT:    ret
+    %3 = call <16 x i8> @llvm.abs.v16i8(<16 x i8> %0, i1 0)
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/abs.ll:128]: Use named values in tests

File: llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp [hunk 2]
```diff
@@ -999,8 +999,11 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
...
     ABSActions
         .legalFor({s32, s64});
-  ABSActions
-      .legalFor(PackedVectorAllTypeList)
+  ABSActions.legalFor(PackedVectorAllTypeList)
+      .clampNumElements(0, v8s8, v16s8)
+      .clampNumElements(0, v4s16, v8s16)
+      .clampNumElements(0, v2s32, v4s32)
+      .clampNumElements(0, v2s64, v2s64)
       .lowerIf(isScalar(0));
```
Comment (arsenm) [inline llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1007]: Can simplify the end to just .lower()?

**Review by arsenm — APPROVED** [review]

**Comment by davemgreen** [comment]
Is it worth having a moreElementsToNextPow2 too?

**Review by davemgreen — APPROVED** [review]
Thanks. LGTM

**Pull request merged**